syz-cluster/pkg/db/session_test_repo.go

```diff
@@ -41,7 +41,10 @@ func (repo *SessionTestRepository) InsertOrUpdate(ctx context.Context, test *Ses
 			_, iterErr := iter.Next()
 			if iterErr == nil {
```

Comment:
Reordering this if->else if->else you can merge 2 beforeSave calls and shift left some code

Reply:
```golang
if iterErr != iterator.Done {
	return iterErr
}
if beforeSave != nil {
	err := beforeSave(test)
	if err != nil {
		return err
	}
}
if iterErr == nil {
	...
}
```